lib/validator/common.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2016-2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/api/unversioned"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+	"github.com/projectcalico/libcalico-go/lib/scope"
+)
+
+// ValidateKeyFieldsExist is used to validate the Key fields in a resource
+// to ensure that all necessary fields are present.
+// This is split from the validator so it can be used conditionally
+// depending on the command.
+func ValidateKeyFieldsExist(resource unversioned.Resource) error {
+	return ValidateMetadataFieldsExist(resource.GetResourceMetadata())
+}
+
+// ValidateMetadataFieldsExist is used to validate the Resource Metadata to ensure
+// that all necessary fields are present.
+// This is split from the validator so it can be used conditionally
+// depending on the command.
+func ValidateMetadataFieldsExist(rm unversioned.ResourceMetadata) error {
```

Comment:
The name suggests this is validating all metadata fields exist - which isn't true, so name needs to be tweaked.  Perhaps ValidateMetadataIDsAssigned()?